### PR TITLE
Implementing document.hasFocus method, needs tests. #6475

### PR DESCRIPTION
**Pull request by mskrzypkows**
I'm not sure if I have to implement some test for a new document method. As I remember there were tests for document methods, is it changed now? Where should I add tests?

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6757)

### CODE REVIEW

**Comment by jdm** [comment]
It appears that http://mxr.mozilla.org/servo/source/tests/wpt/web-platform-tests/html/editing/focus/document-level-focus-apis/document-level-apis.html and http://mxr.mozilla.org/servo/source/tests/wpt/web-platform-tests/html/dom/interfaces.html are the only tests that exist for this right now.

**Comment by jdm** [comment]
They can be run with `./mach test-wpt tests/wpt/web-platform-tests/html/editing/focus/document-level-focus-apis/document-level-apis.html` and `./mach test-wpt tests/wpt/web-platform-tests/html/dom/interfaces.html` respectively.

**Comment by mskrzypkows** [comment]
Thanks, I've added some tests.

File: tests/wpt/web-platform-tests/html/editing/focus/document-level-focus-apis/document-level-apis.html [hunk 1]
```diff
@@ -30,5 +30,9 @@
   assert_false(doc.hasFocus(), "The hasFocus() method should return false.");
 }, "The hasFocus() method must return false if the Document has no browsing context");
 
+test(function () {
+  assert_true(document.hasFocus(), "The hasFocus() method should return true.");
```
Comment (mbrubeck) [inline tests/wpt/web-platform-tests/html/editing/focus/document-level-focus-apis/document-level-apis.html]: I'm not sure whether it's always true that the browser has focus while WPT is running.  (Even if it works with the Servo test harness, it might fail for other browsers.)  This may be why this test was not included in this file originally.  It might be safest to revert this change.

Comment (mskrzypkows) [inline tests/wpt/web-platform-tests/html/editing/focus/document-level-focus-apis/document-level-apis.html]: OK, removed.

**Comment by mbrubeck** [comment]
Reviewed 4 of 4 files at r1, 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6757)

**Comment by mbrubeck** [comment]
@bors-servo r+

Thanks!

**Comment by jdm** [comment]
The tests that failed are unrelated (so I filed #6930 and #6929), but before we try again could you squash the two commits into one?

**Comment by mskrzypkows** [comment]
Sorry for the delay, I'll squash them today.
